### ISSUE TITLE
Ads once again causes black screen of death with inputstream.adaptive

### ISSUE THREAD

**Issue by fragande** [anxdpanic/plugin.video.twitch#452]
Duplicate of #439. I hate to bring this up again but about a week ago I got an oauth error and had to generate a new key. After that I started receiving ads again with a black screen of death after the ad. I'm guessing the workaround of changing platform has been patched by Twitch.

I'm not sure how to interpret #447. Is a fix allowing ads to play normally with inputstream.adaptive planned for the future?

**Comment by anxdpanic** [comment]
Yes, the intention is to have ads working as they should in inputstream.adaptive in the future.
Will update here when there is some movement on the issue.

**Comment by gawdsed** [comment]
running twitch addon on the RPI4 w/libreelec, I'm getting straight up a black screen, is this the same issue?

**Comment by gawdsed** [comment]
[log-2019-08-23-01.58.42.zip](https://github.com/MrSprigster/Twitch-on-Kodi/files/3532540/log-2019-08-23-01.58.42.zip)

**Comment by M-Reimer** [comment]
Does someone have details how they serve the ad stuff? Can I just filter this out using a hosts file?

**Comment by gawdsed** [comment]
not sure if this helps, but I used to run pi-hole a few months back which would stop all ads... which is DNS blocking. I can't say for sure that's still the same method they use.

**Comment by fragande** [comment]
> 
> 
> Does someone have details how they serve the ad stuff? Can I just filter this out using a hosts file?

Unfortunately not. They now inject ads into the stream directly using "EXT-X-DISCONTINUITY" tags. The ad is therefore part of the stream (same URL) which makes it impossible to block with hosts file or DNS blocking like PiHole. Streamlink has managed to implement a way of skipping the segments somehow, but @anxdpanic has stated that he does not intend to implement a way of skipping these ads as it would be in bad faith (which I can respect).

> 
> 
> not sure if this helps, but I used to run pi-hole a few months back which would stop all ads... which is DNS blocking. I can't say for sure that's still the same method they use.

See above. It's not possible to DNS block these ads unfortunately. The ad segments needs to be ignored/skipped by software as they're served from the same URL.
